src/main/scala/tech/cryptonomic/conseil/Conseil.scala

```diff
@@ -31,115 +42,157 @@ object Conseil
     with ConseilOutput {
 
   loadApplicationConfiguration(args) match {
-    case Right((server, platforms, securityApi, verbose, metadataOverrides, nautilusCloud)) =>
+    case Left(errors) =>
+    //nothing to do
+    case Right((server, platforms, securityApi, failFast, verbose, metadataOverrides, nautilusCloud)) =>
       implicit val system: ActorSystem = ActorSystem("conseil-system")
       implicit val materializer: ActorMaterializer = ActorMaterializer()
       implicit val executionContext: ExecutionContextExecutor = system.dispatcher
-      nautilusCloud.foreach { ncc =>
-        system.scheduler.schedule(ncc.delay, ncc.interval)(Security.updateKeys(ncc))
-      }
 
-      val validateApiKey: Directive[Tuple1[String]] = optionalHeaderValueByName("apikey").tflatMap[Tuple1[String]] {
-        apiKeyTuple =>
-          val apiKey = apiKeyTuple match {
-            case Tuple1(apiKey) => apiKey
-            case _ => None
-          }
+      val retries = if (failFast.on) Some(0) else None
+
+      val serverBinding =
+        retry(maxRetry = retries, deadline = Some(server.startupDeadline fromNow))(
+          initServices(server, platforms, metadataOverrides, nautilusCloud).get
+        ).andThen {
+          case Failure(error) =>
+            logger.error(
+              "The server was not started correctly, I failed to create the required Metadata service",
+              error
+            )
+            Await.ready(system.terminate(), 10.seconds)
```

Comment:
Why is this hard coded to 10 seconds?

Reply:
The system is closing anyway, so it's just to give it some room to cleanly close any resource taken, but generally I don't think that anything bad might happen if we don't wait. The Jvm process would be killed anyway.

I think the system will usually be terminated in far less, or will hang on until forced because something is "stuck".
E.g. Lorre's process usually can't close all the streams in a timely manner, but waiting longer won't resolve the thing anyway as far as my trial showed.

tl;dr; I don't consider this value a critical one, it's simply an arbitrary time needed by the code. Making it configurable won't actually change much in my understanding.